### DISCUSSION TITLE
Is there a way to make the "View" keymaps work from the "Object-broswer"

### DISCUSSION THREAD

**Discussion by caeu** [R-nvim/R.nvim#213] [General]
Specifically the function for those keymaps:
```
RViewDFa      n    <localLeader>vh
RViewDFv      n    <localLeader>vv
RViewDFs      n    <localLeader>vs
RDputObj      n    <localLeader>td
RViewDF       n    <localLeader>rv
```

If not, could it be considered for a future feature request.
Thanks

**Comment by PMassicotte** [comment]
This works fine on my side.

![Peek 2024-08-19 09-22](https://github.com/user-attachments/assets/6d8fe09d-5a5b-481d-8617-595e230003c4)

**Reply by caeu** [reply]
Thanks for the quick reply, it works on my side as well, but somehow missed it.

**Reply by caeu** [reply]
I missed it because the table was too large for an old system working on HDDs. I think I will try to set it to a pager (less) and see if it helps. But I guess setting an external pager only works for data.frames and matrices through "csv_app". Is there a way to make the selected pager work for other objects as well, for example large lists?

**Comment by jalvesaq** [comment]
There are Vim plugins and external applications to visualize CSV files. However, lists are better visualized in the Object Browser. You can also type `vi(objname)` on the R Console.

**Reply by caeu** [reply]
A nice and fast tui tool for viewing csv files [csvlens](https://github.com/YS-L/csvlens). It can handle very large tables with some nice features, like regex search/filter by column. I have it configured to work with tmux if r.nvim is running inside tmux, otherwise view with nvim builtin terminal:
```lua
opts = {
   view_df = {
     open_app = os.getenv("TMUX") and "tmux new-window -b -n rnvimview csvlens %s" or "terminal:csvlens"
   }
}
```

I do have a question though. While the default keymap for viewing a dataframes `<localleader>rv` works inside the object browser, I would like to be able to assign one character to open a dataframe from within the object browser, akin to the documentation:
```lua
objbr_mappings = {
  c = 'class',
  l = 'length',
  n = 'names',
  s = 'summary',
  t = 'table',
  v =   ??  --  to display a dataframe in the viewer
},
```
Thanks

**Comment by jalvesaq** [comment]
In this case, we have to copy the code from `lua/r/maps.lua`:

```lua
    objbr_mappings = {
        v = function() require('r.run').action('viewobj') end,
    },
```

**Reply by PMassicotte** [reply]
Can not hurt :)
